.eslintrc.js

```diff
@@ -15,55 +15,50 @@ module.exports = {
     },
   },
   rules: {
-    'import/prefer-default-export': 'off',
     'simple-import-sort/imports': 'warn',
     quotes: [2, 'single', { avoidEscape: true }],
     'prettier/prettier': ['error', { singleQuote: true }],
 
     // TODO: ticket all of these to be enabled or explain why they are disabled
+    // Destructuring props is not required because it clutters the function type signature.
+    // Props can be destructured in a separate line.
     'react/destructuring-assignment': 'off',
+    // Use TypeScript instead of PropTypes for new code.
     'react/forbid-prop-types': 'off',
-    'react/hooks/exhaustive-deps': 'off',
     'react/prop-types': 'off',
     'react/require-default-props': 'off',
     'react/sort-comp': 'off',
     'react/static-property-placement': 'off',
 
-    'import/extensions': 'off',
     'import/named': 'off',
     'import/no-cycle': 'off',
-    'import/no-extraneous-dependencies': 'off',
     'import/no-named-as-default': 'off',
-    'import/no-named-as-default-member': 'off',
-    'import/no-relative-parent-imports': 'off',
+    // Named exports are more convenient for mocking.
+    'import/prefer-default-export': 'off',
 
+    // TS directive comments will be necessary during transition from JS to TS.
     '@typescript-eslint/ban-ts-comment': 'off',
     '@typescript-eslint/ban-types': 'off',
-    '@typescript-eslint/default-param-last': 'off',
-    '@typescript-eslint/naming-convention': 'off',
     '@typescript-eslint/no-empty-function': 'off',
-    '@typescript-eslint/no-explicit-any': ['off'],
-    '@typescript-eslint/no-implied-eval': 'off',
+    '@typescript-eslint/no-explicit-any': 'off',
```

Comment:
lets add a comment that allowing "any" is only during JS to TS transition to allow incremental type improvements.

Reply:
Added in https://github.com/DataBiosphere/terra-ui/pull/3911/commits/9a5283138c1f888b60e11993c276fb0e1dcf1606.